timemachine/fe/chiral_utils.py

```diff
@@ -55,3 +56,69 @@ def setup_chiral_bond_restraints(mol, conf, src_idx, dst_idx):
                 signs.append(-1)
 
     return np.array(restr_idxs), np.array(signs)
+
+
+def find_chiral_atoms(mol):
+    """
+    Find chiral atoms in a molecule. Note that an atom is chiral if it has an non-invertible
+    energy barrier. Even something like CH4 is considered chiral.
+
+    Parameters
+    ----------
+    mol: Chem.Mol
+        input molecule
+
+    Returns
+    -------
+    set of int
+        Chiral atoms
+
+    """
+    # these should be mutually exclusive, but if any pattern is hit then the results
+    # are accumulated to a set
+    chiral_patterns = [
+        "[X4:1]",  # any tetrahedral atom
+        "[#16X3,#15X3:1]",  # trivalent sulfur, phosphorous are assumed to be non-invertible
+        "[#7X3:1](~[R])(~[R])~[R]",  # nitrogen directly bonded to three ring atoms
+    ]
+
+    chiral_atoms = set()
+    for patt in chiral_patterns:
+        query_mol = Chem.MolFromSmarts(patt)
+        assert query_mol is not None
+        for match in mol.GetSubstructMatches(query_mol):
+            chiral_atoms.add(match[0])
+
+    return chiral_atoms
+
+
+def find_chiral_bonds(mol):
+    """
+    Find chiral bonds in a molecule:
+
+    Parameters
+    ----------
+    mol: Chem.Mol
+        input molecule
+
+    Returns
+    -------
+    set of 2-tuple
+        Chiral bonds
+
+    """
+
+    chiral_patterns = [
+        "[X2,X3:1]:[X2,X3:2]",  # any aromatic bond
```

Comment:
I think it should be safe, since I couldn't come up with any examples of X1 or X4 aromatic atoms. I didn't think too hard about this on the initial pass and just went with X2/X3